resources/files/facter.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+unset LD_PRELOAD
+unset LD_LIBRARY_PATH
```

Comment:
Why are we modifying `LD_LIBRARY_PATH` prior to executing any of Puppet's executables? This feels like it could have side effects in a user's environment and be very surprising to them, especially if they have their own LD_LIBRARY_PATH settings in place?